presto-main/src/main/java/com/facebook/presto/security/FileBasedSystemAccessControl.java

```diff
@@ -103,6 +111,29 @@ public SystemAccessControl create(Map<String, String> config)
     @Override
     public void checkCanSetUser(Principal principal, String userName)
     {
+        if (principal == null) {
```

Comment:
Why so.
It seems we should deny here unless `usernameExtractionRules` is empty.

Reply:
That was initially put to avoid any problems for people which would come without authentication and had the file system access control. Indeed we can make this condition tighter.